contracts/ColonyNetworkStaking.sol

```diff
@@ -122,195 +123,4 @@ contract ColonyNetworkStaking is ColonyNetworkStorage, DSMath {
-      nHashesCompletedChallengeRound[round] += 1;
-      // TODO: DRY with the code below.
-      // Check if the hash we just moved to the next round is the second of a pairing that should now face off.
-      nInNextRound = disputeRounds[round+1].length;
-
-      if (nInNextRound % 2 == 0) {
-        disputeRounds[round+1][nInNextRound-1].challengeStepCompleted = 0;
-        disputeRounds[round+1][nInNextRound-1].lastResponseTimestamp = now;
-        disputeRounds[round+1][nInNextRound-2].challengeStepCompleted = 0;
-        disputeRounds[round+1][nInNextRound-2].lastResponseTimestamp = now;
-      }
-    } else {
-      require(disputeRounds[round].length > opponentIdx);
-      require(disputeRounds[round][opponentIdx].hash!="");
-      // Require that it has failed a challenge (i.e. failed to respond in time)
-      require(now - disputeRounds[round][idx].lastResponseTimestamp >= 600); //'In time' is ten minutes here.
-
-      if (disputeRounds[round][opponentIdx].challengeStepCompleted > disputeRounds[round][idx].challengeStepCompleted) {
-        // If true, then the opponent completed one more challenge round than the submission being invalidated, so we
-        // don't know if they're valid or not yet. Move them on to the next round.
-        disputeRounds[round+1].push(disputeRounds[round][opponentIdx]);
-        delete disputeRounds[round][opponentIdx];
-        // TODO Delete the hash(es) being invalidated?
-        nInvalidatedHashes += 1;
-        // Check if the hash we just moved to the next round is the second of a pairing that should now face off.
-        nInNextRound = disputeRounds[round+1].length;
-        if (nInNextRound % 2 == 0) {
-          disputeRounds[round+1][nInNextRound-1].challengeStepCompleted = 0;
-          disputeRounds[round+1][nInNextRound-1].lastResponseTimestamp = now;
-          disputeRounds[round+1][nInNextRound-2].challengeStepCompleted = 0;
-          disputeRounds[round+1][nInNextRound-2].lastResponseTimestamp = now;
-        }
-      } else {
-        // Our opponent completed the same number of challenge rounds, and both have now timed out.
-        nInvalidatedHashes += 2;
-        // Punish the people who proposed our opponent
-        IColonyNetwork(colonyNetworkAddress).punishStakers(submittedHashes[disputeRounds[round][opponentIdx].hash][disputeRounds[round][opponentIdx].nNodes]);
-      }
-
-      // Note that two hashes have completed this challenge round (either one accepted for now and one rejected, or two rejected)
-      nHashesCompletedChallengeRound[round] += 2;
-
-      // Punish the people who proposed the hash that was rejected
-      IColonyNetwork(colonyNetworkAddress).punishStakers(submittedHashes[disputeRounds[round][idx].hash][disputeRounds[round][idx].nNodes]);
-
-    }
-    //TODO: Can we do some deleting to make calling this as cheap as possible for people?
-  }
-
-
-
-}
+// TODO: Can we handle a dispute regarding the very first hash that should be set?
```

Comment:
This comment is in `ReputationMiningCycle` contract already. Does it need to be here too?

Reply:
Bad copy and paste, I'll put it at the top of the network staking contract since it's feature related.